x-pack/server/lib/audit_logger.js

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class AuditLogger {
+  constructor(server, pluginId) {
+    this._server = server;
+    this._pluginId = pluginId;
+  }
+
+  log(eventType, message, data = {}) {
```

Comment:
__question:__ what about `'debug', 'info', 'warning', 'error', 'fatal'` tags?

As far as I understand `eventType` is going to be used for custom event types like `authentication_success`, so there is no way to mark log event as `info` or `error` (it seems some logging verbosity settings like `logging.quiet` rely on these tags), what is plan here? Should `AuditLogger` have public `debug`, `info`, `warn`, `error`, `fatal` methods?

Reply:
That's a great question, and one that I struggled with as well. I couldn't think of any situation where we'd want to `debug`/`info`/`warn`/etc. when doing audit logging, it's almost as if `audit` is it's own eventType. Currently, the only way to get these audit events in the Kibana log is to set `logging.verbose: true`.

One thing I've been thinking about is whether we should create `audit` as a sibling of `debug`/`info`/`warn` and add a new setting `logging.audit: bool` that would enable these item specifically being logged. Thoughts?

Reply:
I agree, I don't see a benefit to having `debug`/`info`/`warn`/etc for audit logging. That being said, I do think it'd be beneficial to offer event filtering, similar to what [ES provides](https://www.elastic.co/guide/en/x-pack/current/auditing.html#audit-log-ignore-policy).

> One thing I've been thinking about is whether we should create audit as a sibling of debug/info/warn...

Having `audit` as a sibling isn't a bad idea, but it feels like this is more of a side effect/implementation detail caused by sharing the underlying log infrastructure with the rest of Kibana.

> ...add a new setting logging.audit: bool that would enable these item specifically being logged. 

How do you feel about keeping the configuration [consistent with ES](https://www.elastic.co/guide/en/x-pack/current/auditing.html#auditing), and naming it `xpack.security.audit.enabled`? -- This might be outside the scope of the base audit logger, however, and could be the responsibility of the Security plugin? ES's use of `xpack.security.audit.enabled` makes it seem like the Security plugin is solely responsible for audit logging, but this base audit logger is decidedly outside of our Security plugin. 
So I guess what I'm getting at is: Do we intend for audit logging to be an application-wide service that's exposed for arbitrary plugins to use, or is it something that only Security should concern itself with?

Reply:
Makes sense - thanks for clarifying!

Reply:
After further pondering the situation, and considering the great points that @legrego brought up. I went ahead and added the `audit` event to the list of default events that will be logged. 

If it's our intent to make users set `xpack.security.audit.enabled: true` or `xpack.beats.audit.enabled: true` to get these audit events in the logs, then it seems right to have them show up in the logs by default, or when `logging.verbose: true`, but not when `logging.quiet: true` or `logging.silent: true`.

Reply:
Good points, thanks. Yeah, indeed, it feels like info/warn/etc separation is irrelevant to audit logging.

Just wanted to note that even though we may have `audit` as a sibling to `info/warn/etc` in the current platform, in the new platform we'll likely use `info` under the hood as [it's done by Elasticsearch](https://github.com/elastic/elasticsearch/blob/72f57c8e72500cb27dc69e902edf5cd270249c14/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java#L180-L191) right now, but you don't have to worry about that.

Reply:
@azasypkin interesting... perhaps we should follow suit and just include the `info` tag so that is' picked up by the default logger when we turn on the plugin specific audit logs...